File: src/Viasfora.LanguageServices/Viasfora.LanguageService.Basic/MethodOverloadsTagger/MethodOverloadsTaggerProvider.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.Text
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Text.Tagging
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition
Imports Winterdom.Viasfora.Contracts
Imports Winterdom.Viasfora.Languages
Imports Winterdom.Viasfora.LanguageService.Core.MethodOverloadsTagger

Namespace MethodOverloadsTagger

  <Export(GetType(IViewTaggerProvider))>
  <ContentType(ContentTypes.VB)>
  <TagType(GetType(MethodOverloadsTag))>
  Public Class MethodOverloadsTaggerProvider
    Inherits BaseMethodOverloadsTaggerProvider

    <ImportingConstructor>
    Public Sub New(
      telemetry As IVsfTelemetry,
      classificationRegistry As IClassificationTypeRegistryService,
      languageFactory As ILanguageFactory,
      settings As IVsfSettings
    )
      MyBase.New(telemetry, classificationRegistry, languageFactory, settings)
    End Sub

    Public Overrides Function IsMethodOverload(node As SyntaxNode) As TextSpan
      If Not SyntaxHelper.IsMethodOverload(node) Then
        Return New TextSpan()
      End If

      Return node.FullSpan
    End Function
  End Class

End Namespace

File: src/Viasfora.LanguageServices/Viasfora.LanguageService.CSharp/MethodOverloadsTagger/MethodOverloadsTaggerProvider.cs
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.Text;
using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Text.Tagging;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;
using Winterdom.Viasfora.Languages;
using Winterdom.Viasfora.LanguageService.Core.MethodOverloadsTagger;

namespace Winterdom.Viasfora.LanguageService.CSharp.MethodOverloadsTagger {
  [Export(typeof(IViewTaggerProvider))]
  [ContentType(ContentTypes.CSharp)]
  [TagType(typeof(MethodOverloadsTag))]
  public sealed class MethodOverloadsTaggerProvider : BaseMethodOverloadsTaggerProvider {

    [ImportingConstructor]
    public MethodOverloadsTaggerProvider(
      IClassificationTypeRegistryService classificationRegistry,
      ILanguageFactory languageFactory,
      IVsfSettings settings
    ) : base(classificationRegistry, languageFactory, settings) {
    }

    public override TextSpan IsMethodOverload(SyntaxNode node) {
      if ( !SyntaxHelper.IsMethodOverload(node) )
        return new TextSpan();

      return node.FullSpan;
    }
  }
}
